cashctrl_ledger/ledger.py

```diff
@@ -181,35 +180,15 @@ def add_account(self, account: str, currency: str, text: str, group: str, vat_co
             group (str): The category group to which the account belongs.
             vat_code (str, optional): The VAT code to be applied to the account, if any.
         """
-
-        currencies = pd.DataFrame(self._client.get("currency/list.json")['data'])
-        currency_map = currencies.set_index('text')['id'].to_dict()
-        if currency not in currency_map:
-            raise ValueError(f"Currency '{currency}' does not exist.")
-        currency_id = currency_map[currency]
-
-        tax_id = None
-        if vat_code is not None:
-            tax_data = self._client.list_tax_rates()
-            tax_map = tax_data.set_index('text')['id'].to_dict()
-            if vat_code not in tax_map:
-                raise ValueError(f"VAT code '{vat_code}' does not exist.")
-            tax_id = tax_map[vat_code]
-
-        categories = self._client.list_categories('account')
-        categories_map = categories.set_index('path')['id'].to_dict()
-        if group not in categories_map:
-            raise ValueError(f"Group '{group}' does not exist.")
-        category_id = categories_map[group]
-
         payload = {
             "number": account,
-            "currencyId": currency_id,
+            "currencyId": self._client.currency_to_id(currency),
             "name": text,
-            "taxId": tax_id,
-            "categoryId": category_id,
+            "taxId": self._client.tax_code_to_id(vat_code, allow_missing=True),
```

Comment:
I believe `vat_code=None` is not covered. 
If `vat_code is not None`, the tax code has to exist.

I suggest
```python
    "taxId": None if vat_code is None else self._client.tax_code_to_id(vat_code),
```

Instead of:
```python
    "taxId": self._client.tax_code_to_id(vat_code, allow_missing=True),
```

Reply:
Same for `update_account()`.